ntpclient/ntpclient.go

```diff
@@ -38,7 +37,10 @@ func NTPClient(pool []string) (time.Time, error) {
 
 		defer con.Close()
 
-		con.SetDeadline(time.Now().Add(5 * time.Second))
+		err = con.SetDeadline(time.Now().Add(5 * time.Second))
```

Comment:
Please change to 
```
if err := con.SetDeadline(time.Now().Add(5 * time.Second)); err != nil {
	log.Warnf(log.TimeMgr, "Unable to SetDeadline. Error: %s\n", err)
        con.Close()
	continue
}
```

The other thing I noticed with this is we are deferring con.Close() but on each for loop iteration, we are reassigning con. The defer will only execute upon func exit, so it's possible for lingering cons